src/main/java/com/exactpro/th2/fix/client/Main.java

```diff
@@ -258,14 +258,16 @@ public static void run(Settings settings, MessageRouter<MessageGroupBatch> messa
 
                             FixMessageFactory messageFactory = (FixMessageFactory) session.getMessageFactory();
 
-                            fixMessage = messageFactory.create(sessionID.getBeginString(), MessageUtils.getMessageType(strMessage), dataDictionary.getOrderedFields());
-                            fixMessage.fromString(strMessage, dataDictionary, true);
+                            qfjMessage = messageFactory.create(sessionID.getBeginString(), MessageUtils.getMessageType(strMessage), dataDictionary.getOrderedFields());
+                            qfjMessage.fromString(strMessage, dataDictionary, true);
                         } else {
-                            fixMessage = MessageUtils.parse(session, strMessage);
+                            qfjMessage = MessageUtils.parse(session, strMessage);
                         }
-                        FixMessage fixMessage1 = new FixMessage(fixMessage, message.getRawMessage().getParentEventId());
-                        if (!session.send(fixMessage1)) {
-                            throw new IllegalStateException("Message not sent. Message was not queued for transmission to the counterparty");
+                        if (!message.getRawMessage().getParentEventId().getId().equals("")) {
+                            FixMessage fixMessage = new FixMessage(qfjMessage, message.getRawMessage().getParentEventId());
+                            sendMessage(session, fixMessage.message);
```

Comment:
> fixMessage.message

We should pass `fixMessage` itself in order for custom `toString()` method to be called